lib/validation.js

```diff
@@ -217,6 +217,26 @@ function validateAsyncHeaders (validatePromise, context, request) {
     })
 }
 
+function validateAsyncHooks (name, fn) {
```

Comment:
I would keep the old code.
It may seem ugly, but it is fast and with a low memory footprint

https://backend.cafe/the-secrets-behind-high-performance-with-nodejs#heading-silly-code-is-not-so-silly